lib/experimental/blocks.php

```diff
@@ -100,3 +100,77 @@ function gutenberg_register_metadata_attribute( $args ) {
 	return $args;
 }
 add_filter( 'register_block_type_args', 'gutenberg_register_metadata_attribute' );
+
+/**
+ * Auto-insert a block as another block's first or last inner block.
+ *
+ * @param array $parsed_block The block being rendered.
+ */
+function gutenberg_auto_insert_child_block( $parsed_block ) {
+	// TODO: Implement an API for users to set the following two parameters.
+	$block_name     = 'core/comment-template';
+	$block_position = 'last-child';
+
+	if ( $block_name === $parsed_block['blockName'] ) {
+		// parse_blocks( '<!-- wp:avatar {"size":40,"style":{"border":{"radius":"10px"}}} /-->' )[0].
+		$inserted_block = array(
+			'blockName'    => 'core/avatar',
+			'attrs'        => array(
+				'size'  => 40,
+				'style' => array(
+					'border' => array( 'radius' => '10px' ),
+				),
+			),
+			'innerHTML'    => '',
+			'innerContent' => array(),
+		);
+
+		if ( 'first-child' === $block_position ) {
+			array_unshift( $parsed_block['innerBlocks'], $inserted_block );
+			// Since WP_Block::render() iterates over `inner_content` (rather than `inner_blocks`)
+			// when rendering blocks, we also need to prepend the new block to that array.
+			array_unshift( $parsed_block['innerContent'], $inserted_block );
```

Comment:
Thinking a bit more about my https://github.com/WordPress/gutenberg/pull/50103#issuecomment-1571505959, I have a feeling that we should prepend `null` for the block without inner content like in the example presented here. That'd probably be something a bit different when handling a block with an inner block, or a block with an inner HTML (static block?).

Reply:
Yeah, looking at [this example](https://github.com/WordPress/gutenberg/blob/a993bfd8e108561bc2ca37a13415847d87c2019f/packages/block-serialization-spec-parser/shared-tests.js#L258-L264), `null` seems like the right choice here. I'll change it.

> That'd probably be something a bit different when handling a block with an inner block, or a block with an inner HTML (static block?).

Indeed 🤔 We can address that later, though 😊

Reply:
> Yeah, looking at [this example](https://github.com/WordPress/gutenberg/blob/a993bfd8e108561bc2ca37a13415847d87c2019f/packages/block-serialization-spec-parser/shared-tests.js#L258-L264), `null` seems like the right choice here. I'll change it.

https://github.com/WordPress/gutenberg/pull/50103/commits/8e9c832125ab7f25577be64519861232c7d496c4